pkg/bgpv1/agent/controller.go

```diff
@@ -188,12 +174,22 @@ func (c *Controller) Run(ctx context.Context) {
 		})
 	)
 
-	// add an initial signal to kick things off
-	c.Sig.Event(struct{}{})
-
 	l.Info("Cilium BGP Control Plane Controller now running...")
 	for {
 		select {
+		case ev, ok := <-c.CiliumNodeResource.Events(ctx):
+			if !ok {
+				l.Info("LocalCiliumNode resource channel closed")
+				return
```

Comment:
I don't know if the handling here is correct.

Reply:
I am not sure if I followed this concern; can you please elaborate on this?